15_3_Testing_with_Regtest.md

```diff
@@ -0,0 +1,142 @@
+}
+```
+
+After creating a transaction, it has to be confirmed and recorded in the Blockchain, the transaction has to be included in a block.
+Most of the applications require 6 block confirmations to consider the transaction as irreversible. If that is your case, you can mine additional 6 blocks into your Regtest chain:
+```
+$ bitcoin-cli -regtest generate 6
+[
+  "33549b2aa249f0a814db4a2ba102194881c14a2ac041c23dcc463b9e4e128e9f",
+  "2cc5c2012e2cacf118f9db4cdd79582735257f0ec564418867d6821edb55715e",
+  "128aaa99e7149a520080d90fa989c62caeda11b7d06ed1965e3fa7c76fa1d407",
+  "6037cc562d97eb3984cca50d8c37c7c19bae8d79b8232b92bec6dcc9708104d3",
+  "2cb276f5ed251bf629dd52fd108163703473f57c24eac94e169514ce04899581",
+  "57193ba8fd2761abf4a5ebcb4ed1a9ec2e873d67485a7cb41e75e13c65928bf3"
+]
+```
+
+
+## Testing with Regtest
+
+When you are in the Regtest mode, you are able to simulate edge cases and attacks that might happen in the real world, such  as Double Spend.
+We are going to use the package [bitcointest by dgarage](https://github.com/dgarage/bitcointest) to simulate a transaction from one wallet to another, but you can check [their guide](https://www.npmjs.com/package/bitcointest) for more specific attack simulations, such as Double Spend.
+
+First of all, you need to install Node.js, and use the NPM (Node Package Manager) to install `bitcointest`:
+```
+$ curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
+$ sudo apt-get install -y nodejs
+$ npm install -g bitcointest
+```
+
+After installing `bitcointest`, you can create the `test.js` file with the following content:
+```
+$ nano test.js
+```
+```javascript
+const { BitcoinNet, BitcoinGraph } = require('bitcointest');
+net = new BitcoinNet('/usr/local/bin', '/tmp/bitcointest/', 22001, 22002);
```

Comment:
Why not `const net` and `const graph` here? They don't seem to be modified and if they were, it should be `let net` / `let graph`.